wgsl/index.bs

```diff
@@ -12218,6 +12351,33 @@ For each call to one of these functions, a [=trigger/derivative_uniformity=] [=d
 
 If one of these functions is called in non-uniform control flow, then the result is an [=indeterminate value=].
 
+Note: Derivatives are an implicit type of [[#quad-ops|quad operation]].
+Their use does not require the [=extension/subgroups=] extension.
+
+### Subgroup Operations ### {#subgroup-ops}
+
+The [[#subgroup-builtin-functions|subgroup built-in functions]] allow efficient
+communication and computation between the invocations in a subgroup.
+Subgroup operations are single-intruction multiple-thread (SIMT) operations.
+
+Because neighbouring invocations commuicate to produce the results, these
+functions should only be invoked in [=uniform control flow=].
```

Comment:
This is too strong I think?

Aren't there many good uses for non-uniform subgroup ops?
e.g. compaction or workstealing?
It's just that you'll get non-portable results.

If we feel super strongly about uniformity being necessary I'd be tempted to have uniform and non-uniform variants, with uniformity checking being stronger by default on the first, and weaker on the second.

Reply:
So this is controlled by a diagnostic (`subgroup_uniformity`) that we had agreed on the default severity of error in a previous WGSL meeting.  This is the same language we use for derivatives (which have their own control). So the default is for portability. I think I prefer using the diagnostic control instead of duplicated functions.

Reply:
I guess my point is "should only" phrasing is too strong.

How about something along these lines.

> The active invocations in a subgroup communicate to determine the results.  Therefore portability is maximized when these functions are invoked when all invocations are active, i.e. in uniform control flow at a subgroup level.

That allows room for the programmer to do less portable things, but with eyes open.  We're not telling the programmer they are wrong to use the functions in non-uniform flow.

Then _maybe_ later emphasize that the programmer can use diagnostic controls to opt in to stricter or looser checks.